Update use of pivot to accomodate pandas 2.0

Pandas 2.0 was just released with backward breaking changes including changes to pivot that are breaking for us.

```
=========================== short test summary info ============================
FAILED tests/test_IoU.py::test_compute_IoU - TypeError: DataFrame.pivot() takes 1 positional argument but 4 were given
FAILED tests/test_evaluate.py::test_evaluate_image - TypeError: DataFrame.pivot() takes 1 positional argument but 4 were given
FAILED tests/test_evaluate.py::test_evaluate - TypeError: DataFrame.pivot() takes 1 positional argument but 4 were given
FAILED tests/test_evaluate.py::test_evaluate_multi - TypeError: DataFrame.pivot() takes 1 positional argument but 4 were given
FAILED tests/test_evaluate.py::test_evaluate_save_images - TypeError: DataFrame.pivot() takes 1 positional argument but 4 were given
FAILED tests/test_main.py::test_train_single - TypeError: DataFrame.pivot() takes 1 positional argument but 4 were given
FAILED tests/test_main.py::test_train_preload_images - TypeError: DataFrame.pivot() takes 1 positional argument but 4 were given
FAILED tests/test_main.py::test_evaluate - TypeError: DataFrame.pivot() takes 1 positional argument but 4 were given
FAILED tests/test_main.py::test_evaluate_multiple_images - TypeError: DataFrame.pivot() takes 1 positional argument but 4 were given
FAILED tests/test_main.py::test_train_callbacks - TypeError: DataFrame.pivot() takes 1 positional argument but 4 were given
FAILED tests/test_main.py::test_save_and_reload_checkpoint - TypeError: DataFrame.pivot() takes 1 positional argument but 4 were given
FAILED tests/test_main.py::test_save_and_reload_weights - TypeError: DataFrame.pivot() takes 1 positional argument but 4 were given
=========== 12 failed, 65 passed, 2059 warnings in 442.41s (0:07:22) ===========
```

--- a/deepforest/IoU.py
+++ b/deepforest/IoU.py
@@ -85,7 +85,8 @@ def compute_IoU(ground_truth, submission):
                               rtree_index=rtree_index)
 
     # Create cost matrix for assignment
-    matrix = overlap_df.pivot("truth_id", "prediction_id", "area").values
+    matrix = overlap_df.pivot(index="truth_id", columns="prediction_id",
+                              values="area").values
     row_ind, col_ind = linear_sum_assignment(matrix, maximize=True)
 
     # Create IoU dataframe, match those predictions and ground truth, IoU = 0

